src/rollup-tx-states.circom

```diff
@@ -201,18 +208,23 @@ template RollupTxStates() {
 
     // select processor 2 function and key2
     ////////
-    isP2Insert <== isExit*newExit; // processor 2 performs an INSERT
+    // processor2 could perform NOP and UPDATE
+    // processor2 performs NOP either:
+    // - finalFromIdx == 0
+    // - L1 transaction and finalToIdx == 0
+    signal isL1FinalToIdxIsZero;
+    isL1FinalToIdxIsZero <== onChain*finalToIdxIsZero.out;
 
-    P2_fnc0 <== isP2Insert*isFinalFromIdx; // processor 2 performs NOP if fromIdx == 0
-    P2_fnc1 <== (1-isP2Insert)*isFinalFromIdx;
+    P2_fnc0 <== 0;
+    P2_fnc1 <== (1 - nop) * (1 - isL1FinalToIdxIsZero);
 
     component mux2 = Mux2();
     mux2.c[0] <== 0;
     mux2.c[1] <== finalToIdx;
-    mux2.c[2] <== 0;
+    mux2.c[2] <== finalFromIdx;
```

Comment:
This line of code is unreachable, i would add a comment.

The only possibilites are:
0 0
0 1
1 0

Reply:
> This line of code is unreachable, i would add a comment.
> 
> The only possibilites are:
> 0 0
> 0 1
> 1 0

`1 0` is unreachable.
Possibilites are the following: `0 0`, `0 1`, & `1 1`